### PR TITLE
Support null values in BYTES data type.

### PR DESCRIPTION
**Pull request by mayankshriv**
1. Use default null value of byte[0] for BYTES data type.
2. Convert ByteBuffer type (from avro) into byte[] transform step,
instead of field extractor step.

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/data/FieldSpec.java [hunk 1]
```diff
@@ -154,8 +154,8 @@ private static Object getDefaultNullValue(@Nonnull FieldType fieldType, @Nonnull
           return stringDefaultNullValue;
         case BYTES:
           try {
-            return new ByteArray(Hex.decodeHex(stringDefaultNullValue.toCharArray()));
-          } catch (DecoderException e) {
+            return stringDefaultNullValue.getBytes("UTF-8");
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/com/linkedin/pinot/common/data/FieldSpec.java]: This looks not correct, we still need to decode hex right?

File: pinot-core/src/main/java/com/linkedin/pinot/core/util/AvroUtils.java [hunk 2]
```diff
@@ -291,6 +293,14 @@ public static Object transformAvroValueToObject(Object avroValue, FieldSpec fiel
     }
     if (fieldSpec.getDataType() == FieldSpec.DataType.STRING) {
       return avroValue.toString();
+    } else if (fieldSpec.getDataType() == FieldSpec.DataType.BYTES && avroValue instanceof ByteBuffer) {
+      // Avro ByteBuffer maps to ByteArray
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/util/AvroUtils.java]: maps to **byte[]**

File: pinot-core/src/main/java/com/linkedin/pinot/core/util/AvroUtils.java [hunk 3]
```diff
@@ -291,6 +293,14 @@ public static Object transformAvroValueToObject(Object avroValue, FieldSpec fiel
...
+      ByteBuffer byteBuffer = (ByteBuffer) avroValue;
+
+      // Assumes byte-buffer is ready to read. Also, avoid getting underlying array, as it may be over-sized.
+      byte[] bytes = new byte[byteBuffer.limit()];
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/util/AvroUtils.java]: Basically the same, but should use **byteBuffer.remaining()**

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java [hunk 4]
```diff
@@ -210,15 +209,6 @@ public GenericRow transform(GenericRow row, GenericRow destinationRow) {
             hasError = true;
             _errorCount.put(column, _errorCount.get(column) + 1);
           }
-        } else if (value instanceof ByteBuffer) { // TODO: Need better handle for ByteBuffers.
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java]: why did we remove this?

Comment (mayankshriv) [reply pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java]: This is moved to transform layer in AvroUtils.

**Pull request merged**